### PR TITLE
B-21432: Display Market Code and I-13434: Market Code Population Correction

### PR DESCRIPTION
**Pull request by TevinAdams**
## Summary

This PR has My work or B-21432 and Daniels work for I-13434

In this PR we are displaying the market code on the shipment card headers in MyMove and Office.

AND

An issue was found where NTS & NTSR shipment market codes weren't being updated on update. I took this opportunity to move the market code logic to the shipments model as well as add an additional helper function for PPM shipments that use a different model type.

All shipments should now be included in the model function and I addressed creation/update of all shipment types.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

## MyMove
1. Access mymove as a customer
2. Create a move or use an existing one
3. Create one or many shipments of various types
4. You should see the market code next to the shipment type on the she shipment cards
Screenshot 2024-10-16 at 9 07 37 AM
image

## Office
1. Access office as any role but Prime
2. Create a move or use an existing one
3. Create one or many shipments of various types
4. You should see the market code next to the shipment type on the she shipment cards

image
image

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## 508 Checks
image
image
image

### CODE REVIEW

File: src/components/Customer/Review/Summary/Summary.jsx [hunk 1]
```diff
@@ -365,6 +370,7 @@ export class Summary extends Component {
           showEditAndDeleteBtn={showEditAndDeleteBtn}
           status={shipment.status}
           onIncompleteClick={this.toggleIncompleteShipmentModal}
+          marketCode={shipment.marketCode}
         />
       );
     });
```
Comment (danieljordan-caci) [inline src/components/Customer/Review/Summary/Summary.jsx:376]: You're missing a change on line `344` in this file from the INT PR

Screenshot 2024-11-04 at 10 23 37 AM

File: src/pages/MyMove/Profile/BackupAddress.test.jsx [hunk 2]
```diff
@@ -111,19 +111,23 @@ describe('BackupAddress page', () => {
...
 
     await userEvent.selectOptions(getByLabelText(/State/), 'HI');
     await userEvent.type(getByLabelText(/ZIP/), fakeAddress.postalCode);
     await userEvent.tab();
-    msg = screen.getByText('Moves to this state are not supported at this time.');
-    expect(msg).toBeVisible();
+    await waitFor(() => {
+      expect(screen.queryByText('Moves to this state are not supported at this time.')).toBeInTheDocument();
+    });
   });
```
Comment (danieljordan-caci) [inline src/pages/MyMove/Profile/BackupAddress.test.jsx:131]: Some differences in this file

`MAIN`
Screenshot 2024-11-04 at 10 26 05 AM

`INT`
Screenshot 2024-11-04 at 10 26 24 AM

File: src/pages/PrimeUI/CreateSITExtensionRequest/CreateSITExtensionRequest.test.jsx [hunk 3]
```diff
@@ -33,7 +33,13 @@ const moveTaskOrder = {
...
       shipmentType: 'HHG',
       requestedPickupDate: '2021-11-26',
-      pickupAddress: { streetAddress1: '100 1st Avenue', city: 'New York', state: 'NY', postalCode: '10001' },
+      pickupAddress: {
+        streetAddress1: '100 1st Avenue',
+        city: 'New York',
+        state: 'NY',
+        postalCode: '10001',
+      },
+      marketCode: 'd',
```
Comment (danieljordan-caci) [inline src/pages/PrimeUI/CreateSITExtensionRequest/CreateSITExtensionRequest.test.jsx:42]: Seems to just be a formatting issue here, in INT it's all inline:
```
pickupAddress: { streetAddress1: '100 1st Avenue', city: 'New York', state: 'NY', postalCode: '10001' },
```
and here you have it lined out:
```
      pickupAddress: {
        streetAddress1: '100 1st Avenue',
        city: 'New York',
        state: 'NY',
        postalCode: '10001',
      },
```

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]

File: src/components/Customer/Review/Summary/Summary.jsx (hunk 1 above)
Comment (TevinAdams) [reply src/components/Customer/Review/Summary/Summary.jsx:377]: I know where this came from. That change should be in here now.

File: src/pages/PrimeUI/CreateSITExtensionRequest/CreateSITExtensionRequest.test.jsx (hunk 3 above)
Comment (TevinAdams) [reply src/pages/PrimeUI/CreateSITExtensionRequest/CreateSITExtensionRequest.test.jsx]: Removed the formatting change.

File: src/pages/MyMove/Profile/BackupAddress.test.jsx (hunk 2 above)
Comment (TevinAdams) [reply src/pages/MyMove/Profile/BackupAddress.test.jsx:131]: This change is coming from you PR https://github.com/transcom/mymove/pull/14000/files#diff-07f83427ecace3e2c2d28e1f16c27f1bd3c38a4efd07162b749c35f4f08bdcff

**Review by danieljordan-caci — APPROVED** [review]
All looks good now. Some minor differences in INT that seem to already be in `main`

**Review by cameroncaci — APPROVED** [review]

**Comment by danieljordan-caci** [comment]
Conflicts - I can do Happo check once you've resolved.

**Review by traskowskycaci — APPROVED** [review]

**Pull request closed without merging**